### ISSUE TITLE
new version of ed.Empirical

### ISSUE THREAD

**Issue by icypriest** [tensorflow/probability#183]
[ed.Empirical](http://edwardlib.org/api/ed/models/Empirical) is deprecated in TensorFlow probability. I've looked through the API and found nothing. Is there a replacement in the new version, or I have to find them elsewhere?

**Comment by davmre** [comment]
Hi @icypriest; that's correct, we don't currently have an Empirical distribution in TFP. The Edward implementation (https://github.com/blei-lab/edward/blob/master/edward/models/empirical.py) is already a TF Distribution, so I think it'd be pretty straightforward to port over to the TFP repo. We'd welcome this as a PR!

**Comment by euler16** [comment]
@davmre can I work on this?

**Comment by brianwa84** [comment]
Glancing at Empirical, I wonder if it could simply subclass MixtureSameFamily(mixture_distribution=Categorical(logits=0), components_distribution=DeterministicNd(samples, event_ndims=X)).
DeterministicNd would be new, but also useful.

**Comment by davmre** [comment]
@brianwa84: that's a cute observation! I bet that would work. DeterministicNd would be a straightforward extension of Deterministic and VectorDeterministic (https://github.com/tensorflow/probability/blob/master/tensorflow_probability/python/distributions/deterministic.py).

@euler16: sure! If you want to try exploring Brian's suggestion to subclass MixtureSameFamily, I think that'd be a great route -- you could check out, e.g., MultivariateNormalDiag for an example of a distribution implemented as a subclass of another distribution -- though a straight-up port could also work. Either way we'd be happy to review a PR. Aside from basic functionality, the main things we look for are solid [unit tests](https://github.com/tensorflow/probability/blob/master/UNITTEST.md), and compliance with the [style guide](https://github.com/tensorflow/probability/blob/master/STYLE_GUIDE.md).

**Comment by joeyhaohao** [comment]
Hi @davmre, I add a PR #191 to address this with Edward Empirical distribution. Would you take a look and give some feedback?

**Comment by jvdillon** [comment]
+1 to MixtureSameFamily(mixture_distribution=Categorical(logits=0), components_distribution=DeterministicNd(samples, event_ndims=X))

**Comment by davmre** [comment]
Thanks for the PR @joeyhaohao!  I went ahead and left some initial feedback. I agree with @jvdillon and @brianwa84 that the MixtureSameFamily representation is elegant and worth exploring, but porting the direct implementation from Edward is a great start (we can always sub in MixtureSameFamily under the hood later on).

**Comment by jvdillon** [comment]
sgtm (so long as we impl it as a TFD then wrap this in E2)
